rtil/src/native/hud.rs

```diff
@@ -116,6 +116,25 @@ fn draw_hud(args: &mut Args) {
     ue::draw_hud();
 }
 
+#[rtil_derive::hook_before(AHUD::DrawMaterialSimple)]
+fn draw_material_simple(args: &mut Args) {
+    let (
+        _this, material, screen_x, screen_y, screen_w, screen_h, scale, scale_position
+    ) = unsafe { args.with_this_pointer::<(*mut UObject, *mut UObject, f32, f32, f32, f32, f32, usize)>() };
+    unsafe {
+        let obj = ObjectWrapper::new(material);
+        if obj.name() == "M_Player_Crosshair" {
+            *screen_w = ue::rebo::STATE.lock().unwrap().as_mut().unwrap().reticle_w;
+            *screen_h = ue::rebo::STATE.lock().unwrap().as_mut().unwrap().reticle_h;
+            *scale = ue::rebo::STATE.lock().unwrap().as_mut().unwrap().reticle_scale;
+            *scale_position = ue::rebo::STATE.lock().unwrap().as_mut().unwrap().reticle_scale_position as usize;
```

Comment:
Instead of adding coupling and depending on implementation details of rebo by accessing the `STATE`, you should have global static atomics next to the hook function. That way rebo can set them and the hook can read them, all without adding a dependency from native -> rebo.